examples/model_selection/plot_cv_indices.py

```diff
@@ -0,0 +1,147 @@
+"""
+Visualizing cross-validation behavior in scikit-learn
+=====================================================
+
+Choosing the right cross-validation object is a crucial part of fitting a
+model properly. There are many ways to split data into training and test
+sets in order to avoid model overfitting, to standardize the number of
+groups in test sets, etc.
+
+This example visualizes the behavior of several common scikit-learn objects
+for comparison.
+"""
+
+from sklearn.model_selection import (TimeSeriesSplit, KFold, ShuffleSplit,
+                                     StratifiedKFold, GroupShuffleSplit,
+                                     GroupKFold, StratifiedShuffleSplit)
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Patch
+np.random.seed(1337)
+cmap_data = plt.cm.Paired
+cmap_cv = plt.cm.coolwarm
+
+###############################################################################
+# Visualize our data
+# ------------------
+#
+# First, we must understand the structure of our data. It has 100 randomly
+# generated input datapoints, 5 labels split unevenly across datapoints,
+# and five "groups" split unevenly across datapoints.
```

Comment:
I think 5 labels and 5 groups are no longer the case.

Reply:
FYI, we've decided to change default behavior from 3-fold to 5-fold, so ``n_splits=4`` seems a bit strange. But I'm fine since the example is visualizing cross-validation behavior.

Reply:
cool - the n_splits=4 behavior is partially just for visual purposes (as you mention the figures are already a bit tight, but I wanted to visualize what a few iterations of the CV looked like)